griffon/commands/entities/community_component_registry.py

```diff
@@ -107,35 +104,8 @@ def list_components(ctx, strict_name_search, component_name, **params):
         ] = "link,uuid,purl,nvr,version,type,name,upstreams,related_url,download_url"
 
     session = CommunityComponentService.create_session()
-    params = multivalue_params_to_csv(params)
-
-    logger.debug("starting parallel http requests")
-    component_cnt = session.components.retrieve_list(**params).count
-    logger.debug(component_cnt)
-    if component_cnt < 3000000:
-        with concurrent.futures.ThreadPoolExecutor() as executor:
-            futures = []
-            components = list()
-            for batch in range(0, component_cnt, 1200):
-                params["offset"] = batch
-                params["limit"] = 1200
-                futures.append(
-                    executor.submit(
-                        session.components.retrieve_list,
-                        **params,
-                    )
-                )
-
-            for future in concurrent.futures.as_completed(futures):
-                try:
-                    components.extend(future.result().results)
-                except Exception as exc:
-                    logger.warning("%r generated an exception: %s" % (future, exc))
-
-            data = sorted(components, key=lambda d: d.purl)
-            return cprint(data, ctx=ctx)
-    else:
-        console.warning("Too many components.")
+    data = [c for c in session.components.retrieve_list_iterator_async(max_results=5000, **params)]
```

Comment:
Nitpick no2: `retrieve_list_iterator_async` returns a generator object so it is perfectly fine to just:
```python
list(session.components.retrieve_list_iterator_async(...))
```

and avoid comprehension. (multiple occurrences below)

Reply:
ah did not occur to me - excellent will change